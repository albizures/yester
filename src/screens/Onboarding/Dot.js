
import React from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet } from 'react-native'

import colors from '../../utils/colors'

const Dot = ({ active }) => {
  return <View style={[styles.dot, { backgroundColor: active ? colors.black : colors.gray }]} />
}

Dot.propTypes = {
  active: PropTypes.bool,
}

const styles = StyleSheet.create({
  dot: {
    width: 10,
    height: 10,
    borderRadius: 5,
    marginHorizontal: 5,
  },
})

export default Dot
