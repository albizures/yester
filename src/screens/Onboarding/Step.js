import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'
import { View, Image } from 'react-native'
import { Heading2, Heading4 } from '../../components'

export default class Step extends Component {
  static propTypes = {
    cover: Image.propTypes.source.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    children: PropTypes.node,
  }
  render () {
    const { cover, title, description, children } = this.props
    return (
      <Fragment>
        <View style={{alignItems: 'center', justifyContent: 'flex-start'}}>
          <Image source={cover} style={{marginBottom: 20}} />
          <Heading2 keyName={title} style={{marginHorizontal: 75, marginVertical: 20}} />
          <Heading4 keyName={description} style={{marginHorizontal: 75}} />
        </View>
        {children}
      </Fragment>
    )
  }
}
