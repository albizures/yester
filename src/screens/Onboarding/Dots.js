import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import Dot from './Dot';

const Dots = ({ steps, currentStep }) => (
	<View style={{ flexDirection: 'row', justifyContent: 'center' }}>
		{new Array(steps).fill(null).map((item, index) => (
			<Dot key={index} active={index === currentStep} />
		))}
	</View>
);

Dots.propTypes = {
	steps: PropTypes.number.isRequired,
	currentStep: PropTypes.number.isRequired,
};

export default Dots;
