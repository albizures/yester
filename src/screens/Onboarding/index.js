import React, { Component } from 'react'
import { View } from 'react-native'
// import PropTypes from 'prop-types'

// import Translate from '../../components/Translate'
import Container from '../../components/Container'
import Stepper from '../../components/Stepper'
import Button from '../../components/Button'
import Step from './Step'
import Dots from './Dots'

export default class Onboarding extends Component {
  state = {
    steps: [{
      cover: require('../../assets/300x300.png'),
      title: 'onboarding.step1.title',
      description: 'onboarding.step1.description',
    }, {
      cover: require('../../assets/300x300.png'),
      title: 'onboarding.step2.title',
      description: 'onboarding.step2.description',
    }, {
      cover: require('../../assets/300x300.png'),
      title: 'onboarding.step3.title',
      description: 'onboarding.step3.description',
    }],
  }

  createStep = (customProp, index) => {
    return (
      <Step key={index} {...customProp} />
    )
  }

  getBottomBar = (options) => {
    const { steps } = this.state
    const { next, currentStep, skip } = options

    return (
      <View>
        <Dots steps={steps.length} currentStep={currentStep} />
        {/* TODO use our custom button here */}
        <Button title='onboarding.next' onPress={skip} />
        <Button title='onboarding.skip' onPress={next} />
      </View>
    )
  }

  render () {
    const { steps } = this.state

    console.log(require('../../assets/300x300.png'))
    return (
      <Container>
        <Stepper bottomBar={this.getBottomBar}>
          {steps.map(this.createStep)}
        </Stepper>
      </Container>
    )
  }
}
