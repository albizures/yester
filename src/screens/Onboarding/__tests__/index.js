import React from 'react';
import renderer from 'react-test-renderer';

import Onboarding from '../';
const navigation = {
	navigate: jest.fn(),
};

describe('src/screens/Onboarding', () => {
	it('should render', () => {
		const tree = renderer
			.create(<Onboarding navigation={navigation} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
