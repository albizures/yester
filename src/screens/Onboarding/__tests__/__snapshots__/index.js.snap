// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/Onboarding should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          <Step
            cover={
              Object {
                "testUri": "../../../src/assets/300x300.png",
              }
            }
            description="onboarding.step1.description"
            title="onboarding.step1.title"
          />,
          <Step
            cover={
              Object {
                "testUri": "../../../src/assets/300x300.png",
              }
            }
            description="onboarding.step2.description"
            title="onboarding.step2.title"
          />,
          <Step
            cover={
              Object {
                "testUri": "../../../src/assets/300x300.png",
              }
            }
            description="onboarding.step3.description"
            title="onboarding.step3.title"
          />,
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={[Function]}
      pagingEnabled={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={
        Array [
          Object {
            "onViewableItemsChanged": [Function],
            "viewabilityConfig": undefined,
          },
        ]
      }
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                },
                Object {
                  "height": 1334,
                  "justifyContent": "center",
                  "paddingTop": 0,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/300x300.png",
                }
              }
            />
            <Text>
              We cant't wait to hear your story
            </Text>
            <Text>
              Everyone has a story to tell. We know yours is definitely a really interesting one.
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                },
                Object {
                  "height": 1334,
                  "justifyContent": "center",
                  "paddingTop": 0,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/300x300.png",
                }
              }
            />
            <Text>
              One day, one story
            </Text>
            <Text>
              We'll guide you through daily question that will help us to get the conversation to get started.
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                },
                Object {
                  "height": 1334,
                  "justifyContent": "center",
                  "paddingTop": 0,
                  "width": 750,
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/300x300.png",
                }
              }
            />
            <Text>
              You own your information
            </Text>
            <Text>
              As you, we want to protect your personal information, so you decide when and with who you share it. Always!
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View>
      <View>
        <View
          style={
            Array [
              Object {
                "borderRadius": 5,
                "height": 10,
                "marginHorizontal": 5,
                "width": 10,
              },
              Object {
                "backgroundColor": "#000",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderRadius": 5,
                "height": 10,
                "marginHorizontal": 5,
                "width": 10,
              },
              Object {
                "backgroundColor": "#c3c3c3",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderRadius": 5,
                "height": 10,
                "marginHorizontal": 5,
                "width": 10,
              },
              Object {
                "backgroundColor": "#c3c3c3",
              },
            ]
          }
        />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            Object {
              "borderRadius": 24,
              "borderWidth": 2,
              "height": 50,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
            },
            Object {
              "backgroundColor": "#4241B3",
              "borderColor": "#4241B3",
            },
            Object {
              "width": 300,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            undefined,
            Object {
              "borderRadius": 24,
              "borderWidth": 2,
              "height": 50,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
            },
            Object {
              "backgroundColor": "#4241B3",
              "borderColor": "#4241B3",
            },
            Object {
              "width": 300,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Skip
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
